### DISCUSSION TITLE
Best method to apply patch to all deployments that meet a target

### DISCUSSION THREAD

**Discussion by shiftychris** [fluxcd/flux2#2729] [Q&A] [answered]
Hi there,

What is the best method or direction to go to apply a patch or strategic merge to all deployments via Git repositories.. I seem to be going down a bit of a rabbit hole between legacy documentation and different examples floating around the internet. In this scenario I am attempting to adding in nodeAffinity and tolerations across the board. I attempted to use the example found on the cheat sheet (https://fluxcd.io/docs/cheatsheets/bootstrap/#node-affinity-and-tolerations), updating the target to reflect an application but could not get the patch to change any future deployments (no error or update to the deployment charts). Is this the best route to go?

**Comment by stefanprodan** [comment]
For Helm Release you can use the `postRenderers` section to patch all deployments in a chart. Docs here https://fluxcd.io/docs/components/helm/helmreleases/#post-renderers

**Reply by shiftychris** [reply]
Correct me if I am wrong but the above will apply it to a specific chart. Can this same process be applied to all deployment on all charts/charts marked with a specific label? We are looking at nodeaffinity across all charts for specific environments. 

Looking at the example in the cheatsheet https://fluxcd.io/docs/cheatsheets/bootstrap/#node-affinity-and-tolerations and modifying it to below, should this apply the target to all deployments to namespace default? or does following that cheat sheet only affect flux components?

`    target:
      kind: Deployment
      namespace: default`

**Comment by shiftychris** [comment, marked as answer]
Just for anyone who gets stuck on this and examples get you going down the wrong path: you need to patch the HelmRelease chart sourced via the GitRepository Kustomization rather than patching specific Deployment charts (which in our environment dont exist)

`  patches:
    - patch: |
       - op: add
         path: "/spec/values/tolerations"
         value: 
          - effect: NoSchedule
            key: "kubernetes.azure.com/scalesetpriority"
            operator: Equal
            value: spot
      target:
        kind: HelmRelease
        namespace: default`
